usaspending_api/common/query_with_filters.py

```diff
@@ -930,10 +943,38 @@ class QueryWithFilters:
 
     unsupported_filters = ["legal_entities"]
 
-    @classmethod
-    def _generate_elasticsearch_query(cls, filters: dict, query_type: _QueryType, **options) -> ES_Q:
-        nested_path = options.pop("nested_path", "")
+    def __init__(self, query_type: _QueryType):
+        self.query_type = query_type
+        time_period_obj = None
+        if self.query_type == _QueryType.ACCOUNTS:
+            self.default_options = {"nested_path": "financial_accounts_by_award"}
+        elif self.query_type == _QueryType.TRANSACTIONS:
+            time_period_obj = TransactionSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
+        elif self.query_type == _QueryType.AWARDS:
+            time_period_obj = AwardSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
+        else:
+            time_period_obj = SubawardSearchTimePeriod(
+                default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+            )
 
+        if time_period_obj is not None:
```

Comment:
Subawards do not support the New Awards Only filter, this should be limited to just Awards and Transactions query type